proposal.html

```diff
@@ -485,7 +485,7 @@
 
   <ins class="block">
     <emu-alg>
-      1. If _func_ is a <emu-xref href="#sec-bound-function-exotic-objects">Bound Function exotic object</emu-xref> or a <emu-xref href="#sec-ecmascript-standard-built-in-objects">built-in Function object</emu-xref>, then return an implementation-dependent String source code representation of _func_. The representation must have the syntax of a |NativeFunction|.
+      1. If _func_ is a <emu-xref href="#sec-bound-function-exotic-objects">Bound Function exotic object</emu-xref> or a <emu-xref href="#sec-ecmascript-standard-built-in-objects">built-in Function object</emu-xref>, then return an implementation-dependent String source code representation of _func_. The representation must have the syntax of a |NativeFunction|. Additionally, if _func_ is a <emu-xref href="#sec-well-known-intrinsic-objects">Well-known Intrinsic Object</emu-xref> and the Global Name associated with _func_ has the syntax of a |BindingIdentifier|, the portion of the returned String that would be matched by |BindingIdentifier| must be the same as the Global Name associated with _func_.
```

Comment:
@gibson042 I think you're not realising that implementations can set whatever values they want to those internal slots, as long as the spec doesn't prohibit them. This step ensures that, even if that slot is there and has a String value, it is ignored for those kinds of functions.